lib/kafo/puppet_log_parser.rb

```diff
@@ -9,7 +9,7 @@ def parse(line)
                           when line =~ /^Error:(.*)/i || line =~ /^Err:(.*)/i
                             [:error, $1]
                           when line =~ /^Warning:(.*)/i || line =~ /^Notice:(.*)/i
```

Comment:
```suggestion
                          when line =~ /Skipping because of failed dependencies$/
                            [:error, $1]
                          when line =~ /^Warning:(.*)/i || line =~ /^Notice:(.*)/i
```

Possible alternate approach:

1. When the message specifically ends with 'Skipping because of failed dependencies' we log it with ERROR so that the user sees them all over the place and realizes they must take the issue seriously
2. Lines logged by Puppet with level Warning or Notice that didn't match this pattern could then be handled separately
3. This would need to be paired with a change to the last line of output to the installer log which captures the first occurrence of an ERROR and directs the user to fix the first ERROR and re-run, so that the user doesn't fixate on the wrong ERROR.

If you are strongly opposed to this approach then I'm fine with proceeding, but I think it's worth at least considering the pros and cons.

Reply:
:-1: to parsing the output further. You can get a [structured report](https://puppet.com/docs/puppet/6.18/format_report.html). If you want to present the user with something actionable, that is IMHO the way to go.